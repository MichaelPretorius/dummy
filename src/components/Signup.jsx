import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from './FormInput';
import { SubmitButton } from './CustomButton';
import { signUpUser } from '../redux/auth/authActions';

const INITIAL_STATE = {
  displayName: '',
  email: '',
  password: '',
  confirmPassword: '',
};

const Signup = ({ history, signUpUser }) => {
  const [credentials, setCredentials] = useState(INITIAL_STATE);

  const { displayName, email, password, confirmPassword } = credentials;

  const handleChange = event => {
    const { name, value } = event.target;
    setCredentials(credentials => ({
      ...credentials,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    if (password !== confirmPassword) {
      alert("Passwords don't match!");
      return null;
    }

    await signUpUser(email, password, { displayName }, () => {
      setCredentials(INITIAL_STATE);
      history.push('/');
    });
  };

  return (
    <>
      <h2>Sign Up</h2>
      <form onSubmit={handleSubmit}>
        <FormInput
          required
          type="text"
          label="User Name"
          name="displayName"
          value={displayName}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="email"
          label="Email"
          name="email"
          value={email}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="password"
          label="Password"
          name="password"
          value={password}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="password"
          label="Confirm Password"
          name="confirmPassword"
          value={confirmPassword}
          handleChange={handleChange}
        />
        <SubmitButton>Sign Up</SubmitButton>
      </form>
    </>
  );
};

export default connect(
  null,
  { signUpUser }
)(Signup);
