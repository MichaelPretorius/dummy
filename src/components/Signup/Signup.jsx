import React, { useState } from 'react';
import { connect } from 'react-redux';

import { Form, Button, Label, InputField, Header, Wrapper } from '../../styles';
import { signUpUser } from '../../redux/auth/authActions';

const INITIAL_STATE = {
  displayName: '',
  email: '',
  password: '',
  confirmPassword: '',
};

export const Signup = ({ history, signUpUser }) => {
  const [credentials, setCredentials] = useState(INITIAL_STATE);

  const { displayName, email, password, confirmPassword } = credentials;

  const handleChange = event => {
    const { name, value } = event.target;
    setCredentials(credentials => ({
      ...credentials,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    if (password !== confirmPassword) {
      // alert("Passwords don't match!");
      return null;
    }

    await signUpUser(email, password, { displayName }, () => {
      setCredentials(INITIAL_STATE);
      history.push('/');
    });
  };

  return (
    <>
      <Header center secondary>
        Sign Up
      </Header>
      <Wrapper>
        <Form onSubmit={handleSubmit} alignEnd>
          <Label secondary>
            User Name
            <InputField
              required
              type="email"
              name="displayName"
              value={displayName}
              onChange={handleChange}
            />
          </Label>
          <Label secondary>
            Email
            <InputField
              required
              type="text"
              name="email"
              value={email}
              onChange={handleChange}
            />
          </Label>
          <Label secondary>
            Password
            <InputField
              required
              type="password"
              name="password"
              value={password}
              onChange={handleChange}
            />
          </Label>
          <Label secondary>
            Confirm Password
            <InputField
              required
              type="password"
              name="confirmPassword"
              value={confirmPassword}
              onChange={handleChange}
            />
          </Label>
          <Button submit primary>
            Sign Up
          </Button>
        </Form>
      </Wrapper>
    </>
  );
};

export default connect(
  null,
  { signUpUser }
)(Signup);
