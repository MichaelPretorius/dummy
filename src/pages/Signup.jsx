import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from '../components/FormInput';
import { SubmitButton } from '../components/CustomButton';
import { signUpUser } from '../redux/auth/authActions';

const Signup = ({ history, signUpUser }) => {
  const [credentials, setCredentials] = useState({
    displayName: '',
    firstName: '',
    lastName: '',
    email: '',
    phoneNumber: '',
    password: '',
    confirmPassword: '',
  });

  const {
    displayName,
    firstName,
    lastName,
    email,
    phoneNumber,
    password,
    confirmPassword,
  } = credentials;

  const handleChange = event => {
    const { name, value } = event.target;
    setCredentials(credentials => ({
      ...credentials,
      [name]: value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    if (password !== confirmPassword) {
      alert("Passwords don't match!");
      return null;
    }

    await signUpUser(
      email,
      password,
      {
        firstName,
        lastName,
        phoneNumber,
        displayName,
      },
      () => history.push('/')
    );

    setCredentials({
      displayName: '',
      firstName: '',
      lastName: '',
      email: '',
      phoneNumber: '',
      password: '',
      confirmPassword: '',
    });
  };

  return (
    <>
      <h2>Sign Up</h2>
      <form onSubmit={handleSubmit}>
        <FormInput
          required
          type="text"
          label="User Name"
          name="displayName"
          value={displayName}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="text"
          label="First Name"
          name="firstName"
          value={firstName}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="text"
          label="Last Name"
          name="lastName"
          value={lastName}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="email"
          label="Email"
          name="email"
          value={email}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="tel"
          label="Phone Number"
          name="phoneNumber"
          value={phoneNumber}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="password"
          label="Password"
          name="password"
          value={password}
          handleChange={handleChange}
        />
        <FormInput
          required
          type="password"
          label="Confirm Password"
          name="confirmPassword"
          value={confirmPassword}
          handleChange={handleChange}
        />
        <SubmitButton className="btn-small blue-grey darken-3">
          Sign Up
        </SubmitButton>
      </form>
    </>
  );
};

export default connect(
  null,
  { signUpUser }
)(Signup);
