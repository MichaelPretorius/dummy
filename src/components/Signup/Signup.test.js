import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from './Signup';

describe('Signup component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Signup />);
  });

  test('should render Signup component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
