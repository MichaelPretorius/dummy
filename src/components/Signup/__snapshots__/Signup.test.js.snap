// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component should render Signup component 1`] = `
"<Fragment>
  <Header center={true} secondary={true}>
    Sign Up
  </Header>
  <Wrapper>
    <Form onSubmit={[Function: handleSubmit]} alignEnd={true}>
      <Label secondary={true}>
        User Name
        <InputField required={true} type=\\"email\\" name=\\"displayName\\" value=\\"\\" onChange={[Function: handleChange]} />
      </Label>
      <Label secondary={true}>
        Email
        <InputField required={true} type=\\"text\\" name=\\"email\\" value=\\"\\" onChange={[Function: handleChange]} />
      </Label>
      <Label secondary={true}>
        Password
        <InputField required={true} type=\\"password\\" name=\\"password\\" value=\\"\\" onChange={[Function: handleChange]} />
      </Label>
      <Label secondary={true}>
        Confirm Password
        <InputField required={true} type=\\"password\\" name=\\"confirmPassword\\" value=\\"\\" onChange={[Function: handleChange]} />
      </Label>
      <Button submit={true} primary={true}>
        Sign Up
      </Button>
    </Form>
  </Wrapper>
</Fragment>"
`;
